paimon-core/src/main/java/org/apache/paimon/table/sink/DynamicBucketRow.java

```diff
@@ -26,8 +26,10 @@
 import org.apache.paimon.data.Timestamp;
 import org.apache.paimon.types.RowKind;
 
+import java.io.Serializable;
+
 /** An {@link InternalRow} wraps another {@link InternalRow} with bucket. */
-public class DynamicBucketRow implements InternalRow {
+public class DynamicBucketRow implements InternalRow, Serializable {
```

Comment:
I am concerns about kryo serialization here...

Maybe we can just use `Tuple2<Row, Integer>` in `SparkWrite`?